### PR TITLE
Add `kubetcl patch`

### PR DESCRIPTION
**Pull request by caesarxuchao**
Add `kubetcl patch`, its implementation is the same as `kubectl update --patch`. 

Usage: kubectl patch RESOURCE NAME -p "JSON patch string"

We plan to deprecate `kubectl update --patch` if this one goes in.

ref #10202

@bgrant0607 @jlowdermilk @nikhiljindal @krousey @ghodss

### CODE REVIEW

**Comment by mbforbes** [comment]
@bgrant0607 for triage / initial review

File: docs/man/man1/.files_generated [hunk 1]
```diff
@@ -17,6 +17,7 @@ kubectl-get.1
 kubectl-label.1
 kubectl-logs.1
 kubectl-namespace.1
+kubectl-patch.1
```
Comment (jlowdermilk) [inline docs/man/man1/.files_generated]: Looks like you forgot to include this file in commit.

**Comment by jlowdermilk** [comment]
LGTM, pending doc fix/squash

File: docs/man/man1/.files_generated (hunk 1 above)
Comment (caesarxuchao) [inline docs/man/man1/.files_generated]: Thanks!

**Comment by caesarxuchao** [comment]
Squashed. Waiting for Shippable to pass.

**Comment by bgrant0607** [comment]
Merge conflict. Please rebase.

**Comment by bgrant0607** [comment]
Good enough for now. LGTM

**Comment by bgrant0607** [comment]
Ping

**Comment by caesarxuchao** [comment]
rebased

**Comment by caesarxuchao** [comment]
remove lgtm label. test-cmd is broken after rebase.

**Comment by caesarxuchao** [comment]
Fixed. @bgrant0607, could add back lgtm after shippable turns green? Sorry for the trouble.

**Comment by caesarxuchao** [comment]
@k8s-bot test this please

**Comment by bgrant0607** [comment]
Reassigning to @alex-mohr for another LGTM since it's a companion to #10468

**Comment by a-robinson** [comment]
Needs rebase again

**Comment by caesarxuchao** [comment]
rebased and squashed. Wait to see if Shippable still passes.

**Pull request merged**